### DISCUSSION TITLE
Example.CLI.PersistentConfig example is stuck after Spotify login

### DISCUSSION THREAD

**Discussion by VitoTringolo** [JohnnyCrazy/SpotifyAPI-NET#780] [Q&A] [answered]
Hi!

First of all thank you for your awesome work, I really like this library.
I had no problem using the library to get tracks and all no user related information, now I want to access the user's playlists so I started investigating how to get user access using the Auth package.

The Example.CLI.PersistentConfig seemed to me the right one. I tried to run it but after I managed to successfully login to Spotify the application is stuck, waiting for the redirection.

What am I missing here?

I added the Redirect URI (http://localhost:5000/callback) inside the Spotify Dashboard.

Thanks for your help

**Comment by JohnnyCrazy** [comment]
Hi,

I guess you supplied the correct `SPOTIFY_CLIENT_ID` as environment variable?

Does the browser window say authentication was successful?

**Reply by VitoTringolo** [reply]
Hey thank you for the quick reply!

Yep I supplied the correct client ID!
And yes the browser window say that the authentication was successful. I am also able to read the URL containing the response 'code' from the browser! 
The problem is that the AuthorizationCodeReceived event handler is never called!

Before I forgot to mention that I am running the example on macOS Monterey 12.4.

**Reply by JohnnyCrazy** [reply]
Did you verify that `AuthorizationCodeReceived` is not called by debugging?
Do you see the following line in the console: `INF >> [WebServer] Cleaning up`?

**Reply by VitoTringolo** [reply, marked as answer]
Ooook I managed to solve it!

The only problem was that the port 5000 was already in use :/ 
I'm sorry I didn't check this before!!

And thank you for your time :)
